authenticate-users/.gitignore

```diff
@@ -0,0 +1,2 @@
+vendor
+composer.lock
```

Comment:
I am on the fence because I don't want to be too strict on the requirements here, since it's a sample app. If it's strict  to one set of packages, it could break on the next version  (e.g. `composer.lock` is for PHP 7.2 and so it fails on PHP 7.3). However, if we omit `composer.lock`,  the opposite could happen (e.g. a newer, incompatible version of a package is installed and breaks). 

But if semantic versioning is observed, I think the second is not very likely, so I'd prefer to leave it out. What do you think?